### DISCUSSION TITLE
How to tailer ERA5 climate data to fit CWatM?

### DISCUSSION THREAD

**Discussion by jane-627** [iiasa/CWatM#19] [Q&A] [answered]
Hi, I wonder if you could help me with data problem. I corrected the ERA5 data, changed the variable names (from 'longitude' to 'lon', from 'latitude' to 'lat') and delete the variable time_bnds. The data are attached below. After I put these files into CWatM, and set:
meteomapssamescale = False
usemeteodownscaling = True
then there happened to be an error:
Step         Date   Discharge
 1      01/01/2017 Traceback (most recent call last):
  File "run_cwatm.py", line 39, in <module>
    main(settings, args)
  File "D:\CWatM-new\cwatm\run_cwatm.py", line 232, in main
    success, last_dis = CWATMexe(settingsfile[0])
  File "D:\CWatM-new\cwatm\run_cwatm.py", line 130, in CWATMexe
    stCWATM.run()
  File "D:\CWatM-new\cwatm\management_modules\dynamicModel.py", line 49, in run
    self.step()
  File "D:\CWatM-new\cwatm\management_modules\dynamicModel.py", line 31, in step
    self._model.dynamic()
  File "D:\CWatM-new\cwatm\cwatm_dynamic.py", line 72, in dynamic
    self.readmeteo_module.dynamic()
  File "D:\CWatM-new\cwatm\hydrological_modules\readmeteo.py", line 341, in dynamic
    self.var.Precipitation, self.var.wc2_prec, self.var.wc4_prec = self.downscaling2(self.var.Precipitation, "downscale_wordclim_prec", self.var.wc2_prec, self.var.wc4_prec, downscale=2)
  File "D:\CWatM-new\cwatm\hydrological_modules\readmeteo.py", line 298, in downscaling2
    quot_wc = divideValues(input, wc4)
  File "D:\CWatM-new\cwatm\management_modules\data_handling.py", line 1693, in divideValues
    z = x / y1
ValueError: operands could not be broadcast together with shapes (75,159) (89,189)

After that I change the settings as :
meteomapssamescale = True
usemeteodownscaling = False
then there was an error:
UnboundLocalError: local variable 'check_clim' referenced before assignment

After that I put "check_clim = False" above "if "usemeteodownscaling" in binding:", then there was another error:
Error 105: None has less a different shape than area or ldd

After that I changed the spatial resolution of the data using cdo remapbil function. However, I found the model counted discharge in a day for 24 times, and the computation process ended in 16/01/2017.
![QQ截图20210718154609](https://user-images.githubusercontent.com/80512850/126059723-ad1cf739-24da-4710-a201-387989b490c3.png)

I don't know what to do with my ERA5 data, could you please help me?
P.S. I download the ERA5 data that is called "ERA5-Land_hourly_data_from_1981_to_present", whose spatial resolution is 0.1°

Files attached:
the latest data:
[era5_hurs_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835820/era5_hurs_2017-2020-cor-remap.zip)
[era5_pr_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835821/era5_pr_2017-2020-cor-remap.zip)
[era5_ps_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835822/era5_ps_2017-2020-cor-remap.zip)
[era5_rlds_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835824/era5_rlds_2017-2020-cor-remap.zip)
[era5_rsds_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835825/era5_rsds_2017-2020-cor-remap.zip)
[era5_sfcwind_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835826/era5_sfcwind_2017-2020-cor-remap.zip)
[era5_tas_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835827/era5_tas_2017-2020-cor-remap.zip)
[era5_tasmax_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835828/era5_tasmax_2017-2020-cor-remap.zip)
[era5_tasmin_2017-2020-cor-remap.zip](https://github.com/iiasa/CWatM/files/6835830/era5_tasmin_2017-2020-cor-remap.zip)

**Comment by guillaumot** [comment]
Hi Jane,
Sorry, I did not manage to unzip your nc files
I can advice some checks before to go further:
- Is your mask map in the same coordinates system than meteo data ? Probably wgs84?
- When you are using different resolutions between your mask (5min?) and your meteo data (0.1°=6min), you should receive a message at the beginning saying that meteo data resolution is X times higher than cwatm resolution. Then, the model will downscale the meteo data but I think they should be a multiplier of the finer resolution.

During your second test (failed after 24 days), you used 'cdo remapbil' to rescale the meteo data at 5min, right? Are you using the same origin for the coordinates? Did you check that coordinates are the same between the mask and the meteo data? In any case, this option seems to work during 24 days, so we can expect an error due to the 'interceptcapNC' layer as mentioned in your error message? Did you change this file? It should contain the interception capacity each 10 days of the year (ie 37 values). If not, it can also be due to the fact that the grid defining your interception capacity is different from the grid defined by the mask.

Luca

**Reply by jane-627** [reply]
Hi Luca,
I have solved this problem. It seems that the unit of variable "time" in ERA5 netcdf file remained "hours since xxxx" though I had aggregated the time period of the netcdf files. I modified the unit to "days since xxx" and CWatM is running smoothly now.
Thank you for your help!
Jane

**Comment by guillaumot** [comment, marked as answer]
Hi Jane,
Thank you for letting me know. So, I wish you good luck for the next and hope you have usefull and interesting results!

Best
Luca
